compiler/modules/nback/structure.bal

```diff
@@ -352,30 +352,52 @@ function mappingOrderFields(t:MappingAtomicType mat, string[] fieldNames, bir:Op
 function buildMappingGet(llvm:Builder builder, Scaffold scaffold, bir:MappingGetInsn insn) returns BuildError? {
     bir:Register mappingReg = insn.operands[0];
     bir:StringOperand keyOperand = insn.operands[1];
-    int? fieldIndex = mappingFieldIndex(scaffold.typeContext(), mappingReg.semType, keyOperand);
-    boolean fill;
-    RuntimeFunction rf;
-    llvm:Value k;
-    if fieldIndex == () {
-        fill = insn.name == bir:INSN_MAPPING_FILLING_GET;
-        rf = fill ? mappingFillingGetFunction : mappingGetFunction;
-        k = check buildString(builder, scaffold, keyOperand);
+    var fieldIndexRes = mappingFieldIndex(scaffold.typeContext(), mappingReg.semType, keyOperand);
+    llvm:Value mapping = builder.load(scaffold.address(mappingReg));
+    if fieldIndexRes == () {
+        llvm:Value k = check buildString(builder, scaffold, keyOperand);
```

Comment:
There are no changes needed in this case (when the mapping is not atomic). This code does not seem to me to be an improvement over the previous code.